### PR TITLE
W-12226023: Allow to waive expression required parameters on design time when the value is a property

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

File: runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/ExpressionsInRequiredExpressionsParamsTestCase.java [hunk 1]
```diff
@@ -22,7 +23,7 @@ public class ExpressionsInRequiredExpressionsParamsTestCase extends AbstractCore
 
   @Override
   protected Validation getValidation() {
-    return new ExpressionsInRequiredExpressionsParams(empty());
+    return new ExpressionsInRequiredExpressionsParams(empty(), false);
```
Comment (eze210) [inline runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/ExpressionsInRequiredExpressionsParamsTestCase.java:26]: Testing the false case?

**Review by eze210 — CHANGES_REQUESTED** [review]

File: runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/ExpressionsInRequiredExpressionsParamsTestCase.java (hunk 1 above)
Comment (eze210) [reply runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/ExpressionsInRequiredExpressionsParamsTestCase.java:26]: I was confused with this one: ```ExpressionsInRequiredExpressionsParamsNonPropertyValue(boolean enabled)```

**Review by eze210 — APPROVED** [review]

**Pull request merged**